src/Microsoft.HttpRepl.Tests/HttpStateTests.cs

```diff
@@ -49,6 +53,146 @@ public void GetPathString_MultiplePathSections_CorrectString()
             Assert.Equal(expected, result);
         }
 
+        [Fact]
+        public void GetApplicableContentTypes_NoBaseAddress_ReturnsNull()
+        {
+            HttpState httpState = SetupHttpState();
+
+            httpState.BaseAddress = null;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, string.Empty);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void GetApplicableContentTypes_NoStructure_ReturnsNull()
+        {
+            HttpState httpState = SetupHttpState();
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = null;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, string.Empty);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void GetApplicableContentTypes_NoMethod_ReturnsAll()
+        {
+            HttpState httpState = SetupHttpState();
```

Comment:
nit: move this down to where you set up the httpsState (line 91ish).  It reads smoother because you're not jumping back and forth between which objects you're setting up, and it saves you an extra empty line.

Reply:
Fixed in a973059.